modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessorBuilder.java

```diff
@@ -152,10 +155,21 @@ public ComponentMessageProcessorBuilder<M, P> setTerminationTimeout(long termina
     return this;
   }
 
-  protected ConfigurationProvider getConfigurationProvider() {
-    return parameters.values().stream()
-        .filter(v -> v instanceof ConfigurationProvider)
-        .map(v -> ((ConfigurationProvider) v)).findAny()
-        .orElse(configurationProvider);
+  protected ValueResolver<ConfigurationProvider> getConfigurationProviderResolver() {
+    // Uses the configurationProvider given to the builder if any, otherwise evaluates the parameters.
```

Comment:
how many times is this method called on any given instance? Shouldn't this be cached?

Reply:
This one is called once per call to `createMessageProcessor` on the builders. So I guess it would only make sense to cache it if we are using the same builder to create multiple processors from the same parameters.
Yet it doesn't seem to be a very costly operation. It is a lookup on a map to get the config-ref parameter, then a few `instanceof` checks and at most one construction of a `StaticValueResolver` which seems pretty cheap.
I could easily store the result in the builder instance and reuse it, if you think it makes sense.